realtek: zyxel gs1900-24: split into A1 and B1, use all 128M ram on B1

jwh opened the pull request:
I overlooked this in my tree, my -24 which identifies as a v2 in the bootloader actually has 128M ram, the default bootcmd limits it to 64M however.

Does someone have a true v1 that they can test on?  If it turns out the v1 actually does only have 64M ram, I'll add v2 as another device.

plappermaul commented:
As per https://forum.openwrt.org/t/support-for-rtl838x-based-managed-switches/57875/2438 please make a v2 out of it. See gs1900-24hp:

- dtsi defines 128mb
- v1 has a 64mb downsize included
- v2 not

Thanks.

jwh committed:
realtek: use common dtsi for gs1900-24

As there are actually 2 versions of the GS1900-24 where the only
difference is the amount of ram, use a common dtsi and make the
original A1 model dts include the dtsi and only override the memory size.

Signed-off-by: Joe Holden <jwh@zorins.us>

jwh committed:
realtek: add gs1900-24 B1 device definition

The ZyXEL GS1900-24 B1 is a 24 port switch with two SFP ports, it is
identical to the A1 except for doubling the ram.

Specifications
--------------
* Device:    ZyXEL GS1900-24 B1
* SoC:       Realtek RTL8382M 500 MHz MIPS 4KEc
* Flash:     16 MiB
* RAM:       128 MiB DDR2 SDRAM
* Ethernet:  24x 10/100/1000 Mbps, 2x SFP 100/1000 Mbps
* LEDs:
  * 1 PWR LED (green, not configurable)
  * 1 SYS LED (green, configurable)
  * 24 ethernet port link/activity LEDs (green, SoC controlled)
  * 2 SFP status/activity LEDs (green, SoC controlled)
* Buttons:
  * 1 "RESET" button on front panel (soft reset)
  * 1 button ('SW1') behind right hex grate (hardwired power-off)
* Power:     120-240V AC C13
* UART:      Internal populated 10-pin header ('J5') providing RS232;
             connected to SoC UART through a SIPEX 3232EC for voltage
             level shifting.

* 'J5' RS232 Pinout (dot as pin 1):
  2) SoC RXD
  3) GND
  10) SoC TXD

Serial connection parameters: 115200 8N1.

Installation
------------

OEM upgrade method:

* Log in to OEM management web interface

* Navigate to Maintenance > Firmware > Management

* If "Active Image" has the first option selected, OpenWrt will need to be
  flashed to the "Active" partition. If the second option is selected,
  OpenWrt will need to be flashed to the "Backup" partition.

* Navigate to Maintenance > Firmware > Upload

* Upload the openwrt-realtek-rtl838x-zyxel_gs1900-24-b1-initramfs-kernel.bin
  file by your preferred method to the previously determined partition.
  When prompted, select to boot from the newly flashed image, and reboot
  the switch.

* Once OpenWrt has booted, scp the sysupgrade image to /tmp and flash it:

  > sysupgrade /tmp/openwrt-realtek-rtl838x-zyxel_gs1900-24-b1-squashfs-sysupgrade.bin

U-Boot TFTP method:

* Configure your client with a static 192.168.1.x IP (e.g. 192.168.1.10).

* Set up a TFTP server on your client and make it serve the initramfs
  image.

* Connect serial, power up the switch, interrupt U-boot by hitting the
  space bar, and enable the network:

  > rtk network on

> Since the GS1900-24 B1 is a dual-partition device, you want to keep the
  OEM firmware on the backup partition for the time being. OpenWrt can
  only be installed in the first partition anyway (hardcoded in the
  DTS). To ensure we are set to boot from the first partition, issue the
  following commands:

  > setsys bootpartition 0
  > savesys

* Download the image onto the device and boot from it:

  > tftpboot 0x81f00000 192.168.1.10:openwrt-realtek-rtl838x-zyxel_gs1900-24-b1-initramfs-kernel.bin
  > bootm

* Once OpenWrt has booted, scp the sysupgrade image to /tmp and flash it:

  > sysupgrade /tmp/openwrt-realtek-rtl838x-zyxel_gs1900-24-b1-squashfs-sysupgrade.bin

Signed-off-by: Joe Holden <jwh@zorins.us>

plappermaul commented:
Result is good. Regarding the commit structure some fixes please.

- Give some better commit explanation to the first commit. Like "... there is a v2 ... prepare dts ... by doing this and that ...."
- Combine the latter two commits into one. Give that commit the same description (aka installation guide) as the initial GS1900-24 commit.

Thanks.

jwh commented:
I've copied the original message verbatim as I'm not 100% what the deal is now (ie; from oem firmware after dropping dual partitions etc) so it may or may not be accurate now

plappermaul commented:
@andyboeh might have an additional info here.

plappermaul reviewed (approved):
Otherwise fine for me.

hauke commented:
What does the label on the switch say? If it says version `A` and `B1`, we should use that instead of v1 and v2.

andyboeh commented:
> @andyboeh might have an additional info here.

No, sorry, I don't have either of these switches.

svanheule commented on target/linux/realtek/image/rtl838x.mk:
```diff
@@ -411,6 +411,15 @@ define Device/zyxel_gs1900-24-v1
+  SOC := rtl8382
+  DEVICE_MODEL := GS1900-24
+  DEVICE_VARIANT := v2
+  ZYXEL_VERS := AAHL
```
Without `SUPPORTED_DEVICES += zyxel,gs1900-24-v1` users will have to force upgrade existing installs on B1 hardware. The system configuration could be restored, but you may want to mention this in the commit message.

To be clear, I wouldn't add `SUPPORTED_DEVICES`, as it (hopefully) makes the user think about what they are about to do.

svanheule commented on target/linux/realtek/dts/rtl8382_zyxel_gs1900-24-v2.dts:
```diff
@@ -0,0 +1,8 @@
+
+/ {
+	compatible = "zyxel,gs1900-24-v2", "realtek,rtl838x-soc";
+	model = "Zyxel GS1900-24 v2";
```
Perhaps the "rev. B1" naming could be used here, to match the label?

svanheule commented on target/linux/realtek/dts/rtl8382_zyxel_gs1900-24-v1.dts:
```diff
@@ -1,7 +1,6 @@
+#include "rtl8382_zyxel_gs1900-24.dtsi"
 
 / {
 	compatible = "zyxel,gs1900-24-v1", "realtek,rtl838x-soc";
```
If the v2 gets `model = ... rev. B1`, perhaps this could be updated to say `rev. A1`

svanheule reviewed (approved):
Some non-critical comments (admittedly mostly bikeshedding), otherwise LGTM.

As for the B1/v2 naming, I'm okay with either but the target has historically used vX naming (and gs1900-24-v1 already exists).

GoetzGoerisch commented:
@jwh rebase and addressed @svanheule comment to memory size.
Please feel free to reset/pick to your branch and this PR to get this merged and maybe into 25.12.

b5f23ee472c169f09bdc10b0a8fc2c809180caa0
